File: src/zanzibar.py
for _ in range(int(input())):
    numbers = [int(d) for d in input().split()[:-1]]
    total = 0
    for i in range(1, len(numbers)):
        total += max(numbers[i] - 2 * numbers[i - 1], 0)
    print(total)

File: src/zanzibar.java
import java.util.Scanner;

class StandOnZanzibar {
    public static void main(String[] args) {
        Scanner s = new Scanner(System.in);
        int t = s.nextInt();
        for (int i = 0; i < t; i++) {
            int total = 0;
            int n = 0;
            int init = 1;
            while (true) {
                n = s.nextInt();
                if (n == 0) {
                    break;
                }
                total += Math.max(n - 2 * init, 0);
                init = n;
            }
            System.out.println(total);
        }
    }
}
